Source/Scene/Model.js

```diff
@@ -1870,16 +1916,23 @@ define([
     };
 
     CreateProgramJob.prototype.execute = function() {
-        createProgram(this.id, this.model, this.context);
+        createProgram(this.id, this.model, this.context, true);
     };
 
     ///////////////////////////////////////////////////////////////////////////
 
-    function createProgram(id, model, context) {
-        var programs = model.gltf.programs;
-        var shaders = model.gltf.shaders;
+    // When building programs for the first time, do not include modifiers for clipping planes and color
+    // since this is the version of the program that will be cached.
+    function createProgram(id, model, context, firstBuild) {
+        var programs = model._sourcePrograms;
+        var shaders = model._sourceShaders;
+        var quantizedVertexShaders = model._quantizedVertexShaders;
+
         var program = programs[id];
 
+        var clippingPlaneCollection = model.clippingPlanes;
+        var addClippingPlaneCode = !firstBuild && needsClippingCode(model);
```

Comment:
This needs a similar check for color.

Maybe irrelevant based on the comment above.